easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -918,6 +932,33 @@ def resolve_template(value, tmpl_dict):
 
     return value
 
+def replace_templates(value, templ_const, templ_val):
+    """
+    Given a value, try to substitute template strings where possible.
+        - value can be a string, list, tuple, dict or combination thereof
+        - templ_const is a dictionary of template strings (constants)
+        - templ_val is a dictionary of template strings specific for this easyconfig file
+    """
+    if isinstance(value, basestring):
+        old_value = ""
+        while value != old_value:
```

Comment:
doing the while loop here means you assume that each template value is a string

that's probably true today, but may not be in the future

so, why not make this a recursive function?